protocol/test/forge/TempleTeamPaymentsFactory.t.sol

```diff
@@ -0,0 +1,209 @@
+            recip[i] = test;
+            uint256 testAmount = (i + 1) * 1 ether;
+            values[i] = testAmount;
+            totalPaid += testAmount;
+        }
+
+        vm.startPrank(multisig);
+        temple.approve(address(factory), type(uint256).max);
+
+        bytes32 salt = keccak256(
+            abi.encodePacked(temple, factory.lastPaidEpoch() + 1)
+        );
+        address nextEpochClone = Clones.predictDeterministicAddress(
+            factory.templeTeamPaymentsImplementation(),
+            salt,
+            address(factory)
+        );
+        vm.expectEmit(true, true, true, true);
+        emit FundingDeployed(
+            address(temple),
+            factory.lastPaidEpoch() + 1,
+            recip,
+            values,
+            nextEpochClone
+        );
+        TempleTeamPaymentsV2 testContract = factory.deployPayouts(
+            temple,
+            recip,
+            values,
+            totalPaid
+        );
+        vm.stopPrank();
+
+        for (uint256 i; i < recip.length; i++) {
+            address tester = vm.addr(i + 1);
+            uint256 currentBalance = testContract.allocation(tester);
+            assertEq(currentBalance, values[i]);
+        }
+
+        return testContract;
+    }
+
+    function testDeployPayoutsSingle() public returns (TempleTeamPaymentsV2) {
+        return testDeployPayoutsAssertions(1);
+    }
+
+    function testDeployPayoutsMany() public returns (TempleTeamPaymentsV2) {
+        return testDeployPayoutsAssertions(50);
+    }
+
+    function testRoundIncrementsDirectPayout() public {
+        uint256 prev = factory.lastPaidEpoch();
+        testDirectPayoutsSingle();
+        assertEq(factory.lastPaidEpoch(), prev + 1);
+    }
+
+    function testRoundIncrementsDeployPayout() public {
+        uint256 prev = factory.lastPaidEpoch();
+        testDeployPayoutsSingle();
+        assertEq(factory.lastPaidEpoch(), prev + 1);
+    }
+
+    function testCanClaimAllocation() public {
+        TempleTeamPaymentsV2 testContract = testDeployPayoutsSingle();
+
+        uint256 prev = testContract.temple().balanceOf(testUser);
+        uint256 alloc = testContract.allocation(testUser);
+        vm.expectEmit(true, true, true, true);
+        emit Claimed(address(testUser), alloc);
+        vm.prank(testUser);
+        testContract.claim();
+
+        assertEq(
+            testContract.temple().balanceOf(testUser),
+            prev + testContract.allocation(testUser)
+        );
+    }
+
+    function testCannotClaimTwice() public {
+        TempleTeamPaymentsV2 testContract = testDeployPayoutsSingle();
+
+        vm.startPrank(testUser);
+
+        testContract.claim();
+
+        vm.expectRevert();
+        testContract.claim();
+    }
+
+    function testCannotClaimPause() public {
+        TempleTeamPaymentsV2 testContract = testDeployPayoutsSingle();
+
+        vm.prank(multisig);
+        testContract.toggleMember(testUser);
+
+        vm.expectRevert();
```

Comment:
encode custom error as argument to `expectRevert`. `vm.expectRevert(bytes("CustomError"));` or with 
```
vm.expectRevert(
    abi.encodeWithSelector(MyContract.CustomError.selector, 1, 2)
);
```